### PR TITLE
[Reservations] Upgrade SDK to version 0.6.0

### PR DESCRIPTION
**Pull request by Only2125**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by fengzhou-msft** [comment]
@Only2125 could you please rebase against dev so that travis ci can pass with a latest fix on dev.

**Comment by Only2125** [comment]
Hi, I saw all unit tests pass properly in my local machine. And it passed both in -live and nonlive. Can anybody help me understand why tests are failing in the checks?

File: src/azure-cli/azure/cli/command_modules/reservations/tests/latest/recordings/test_get_catalog.yaml [hunk 1]
```diff
@@ -2,323 +2,120 @@ interactions:
...
+      ParameterSetName:
+      - --subscription-id --reserved-resource-type --location
+      User-Agent:
+      - python/3.7.5 (Windows-10-10.0.18362-SP0) msrest/0.6.10 msrest_azure/0.6.2
+        azure-mgmt-reservations/0.6.0 Azure-SDK-For-Python AZURECLI/2.0.76
+      accept-language:
+      - en-US
     method: GET
-    uri: https://management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/providers/Microsoft.Capacity/catalogs?api-version=2018-06-01&reservedResourceType=SqlDatabases&location=westus
+    uri: https://api-dogfood.resources.windows-int.net/subscriptions/00000000-0000-0000-0000-000000000000/providers/Microsoft.Capacity/catalogs?api-version=2019-04-01&reservedResourceType=VirtualMachines&location=westus
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/reservations/tests/latest/recordings/test_get_catalog.yaml]: This is causing ci issue.

File: src/azure-cli/HISTORY.rst [hunk 2]
```diff
@@ -30,6 +30,11 @@ Release History
 * Support enabling auditing on creating a registry
 * Support Repository-scoped RBAC
 
+**Reservations**
+
+* [BREAKING CHANGE] Operation ReservationOrderOperations.get has a new signature
+* [BREAKING CHANGE] Operation ReservationOperations.get has a new signature
```
Comment (haroldrandom) [inline src/azure-cli/HISTORY.rst]: Just `Upgrade SDKversion to 0.6.0`

**Pull request merged**